style/git/README.md

```diff
@@ -179,9 +195,60 @@ scope: {
+
+1. Create a branch off of master to fix an issue assigned to you.
+
+1. Push this branch up to GitHub, and create a pull request when you are ready.
+
+1. Assign a reviewer to your pull request. Leave some nice documentation or even a [screencast](https://viewedit.com/) to give your reviewer a bit more background about the code you wrote in this pull request (include information about any interesting or difficult code, and where the reviewer can see the code that you are referencing).
+
+1. Your reviewer will either approve or request changes on your pull request.  
+    - If they request changes, please implement or further discuss their comments with them.
+
+1. If your reviewer approves your pull request, they will *rebase* your branch on top of master (see my description of rebasing in the next section for more information).
+
+Rebasing a Branch onto the Master Branch
+----------------------------------------------
+
+Rather than merging commits from another branch into the master branch, Sparkbox prefers to rebase commits on top of the master branch.  Here is a diagram that depicts the differences between merging and rebasing:
+
+![Merging](merge.png)
+
+![Rebasing](rebase.png)
+
+For the purposes of this example explaining rebasing, let's say that the branch you need to rebase is called `chore-teaching-example`, and that this branch has gone through all of the Git flow discussed above for a pull request.  Follow these next steps to rebase a branch onto master:
+
+1. Open your terminal and checkout the branch to be rebased: `git checkout chore-teaching-example`.
+
+1. Make sure you have all of the latest changes for this branch. You can run: `git pull origin chore-teaching-example`, or if this branch has been force-pushed to, you can run: `git pull origin chore-teaching-example --rebase`.
+
+1. Run this branch locally and make sure nothing is broken.
+
+1. _Optional_: run `git show-branch origin/master chore-teaching-example` to see where your branch is in relation to the master branch (is it behind the master branch by a lot of commits?). Ideally, your branch should have all of the commits that the master branch has.
+
+1. If your branch is behind the master branch's commits, run `git rebase origin/master`. You can also run `git rebase origin/master -i` which will allow you to _interactively_ (that's where the `-i` comes from) select which commits you want to be added to the master branch, and also permit you to squash commits and reword commit messages. If your branch is already rebased, you can skip to step 7.
+
+    - Upon rebasing, you may run into merge conflicts.  Follow the helpful instructions offered by Git in your terminal, and fix all of the merge conflicts until you have rebased your branch on top of master.
+
+1. After rebasing your branch on top of the master branch, you'll need to force push these changes to your GitHub version of the branch. Run `git push origin chore-teaching-example --force`.
```

Comment:
@ethanmuller has an awesome tip that you should do this with `--force-with-lease`.